app/routes/circulars/circulars.server.ts

```diff
@@ -379,9 +383,30 @@ export async function putVersion(
     editedBy: formatAuthor(user),
     editedOn: Date.now(),
   }
+
   validateCircular(newCircularVersion)
 
-  return await circularVersionsAutoIncrement.put(newCircularVersion)
+  const promises = [
+    circularVersionsAutoIncrement.put(newCircularVersion),
+  ] as any[]
+
+  if (newCircularVersion.eventId) {
+    promises.push(
+      manageSynonymVersionUpdates(
+        newCircularVersion.eventId,
+        oldCircular.eventId,
+        newCircularVersion.createdOn
+      )
+    )
+  }
+
+  const results = await Promise.all(promises)
+
+  if (!newCircularVersion.eventId && oldCircular.eventId) {
+    await deleteIfGroupIsEmpty(oldCircular.eventId)
+  }
```

Comment:
Is this necessary, given that `manageSynonymVersionUpdates` itself calls `deleteIfGroupIsEmpty`?